plan/physical_plan_builder.go

```diff
@@ -792,6 +1027,11 @@ func (p *Selection) convert2PhysicalPlan(prop *requiredProperty) (*physicalPlanI
 	if info != nil {
 		return info, nil
 	}
+	info = p.tryToBuildScanByKeyAndCorCol()
```

Comment:
This check is very heavy. It may be called multiple times. I suggested that If 'flagDecorrelated' is true, check it in an other logic but in physical plan builder.